pkg/fdbstatus/status_checks.go

```diff
@@ -390,9 +390,26 @@ func DoLogServerFaultDomainCheckOnStatus(status *fdbv1beta2.FoundationDBStatus)
 }
 
 // DoCoordinatorFaultDomainCheckOnStatus does a coordinator related fault domain check over the given status object.
-// @note an empty function for now. We will revisit this later.
-func DoCoordinatorFaultDomainCheckOnStatus(_ *fdbv1beta2.FoundationDBStatus) error {
-	// TODO: decide if we need to do coordinator related check.
+func DoCoordinatorFaultDomainCheckOnStatus(status *fdbv1beta2.FoundationDBStatus) error {
```

Comment:
I don't see how this or parent call is used in the code - but since status update is eventual and we read it from object we can run in situation where current coordinator is not reachable anymore and we assume it's based on this check

Reply:
What do you mean by `status update is eventual`?

> we can run in situation where current coordinator is not reachable anymore and we assume it's based on this check

This can happen even if we fetch the status directly before doing this check e.g. because at the moment we fetched the status the coordinator was still reachable. The question is only how long the interval between the failure and the detection is. In general those checks intention are to reduce the risk of failures, we are not able to prevent any outages, only to reduce the risk of them.